cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreImpl.java

```diff
@@ -1163,18 +1164,22 @@ public Object call(SQLDatabase db) throws Exception {
                         long docNumericId = getNumericIdInQueue(db, item.rev.getId());
                         long seq = 0;
 
+                        InsertRevisionCallable.Result result;
                         if (docNumericId != -1) {
-                            seq = doForceInsertExistingDocumentWithHistory(db, item.rev, docNumericId, item.revisionHistory,
+                             result = doForceInsertExistingDocumentWithHistory(db, item.rev, docNumericId, item.revisionHistory,
                                     item.attachments);
-                            item.rev.initialiseSequence(seq);
                             // TODO fetch the parent doc?
                             documentUpdated = new DocumentUpdated(null, item.rev);
                         } else {
-                            seq = doForceInsertNewDocumentWithHistory(db, item.rev, item.revisionHistory);
-                            item.rev.initialiseSequence(seq);
+                            result  = doForceInsertNewDocumentWithHistory(db, item.rev, item.revisionHistory);
                             documentCreated = new DocumentCreated(item.rev);
                         }
 
+                        item.rev.initialiseSequence(result.sequence);
```

Comment:
Because the sequence is now held in the `InsertRevisionCallable.Result` we no longer use the `long seq` field so that can be deleted.